### PR TITLE
More billing nits

### LINKED ISSUE #390: Add a billing grace period before accounts are blocked
From @andrewxhill regarding Hub billing:
>  Can we flag this for later enhancement to enable a less abrupt outage. I've seen plans start accruing a bill and giving the user 1-week to enable billing and pay or then error the APIs. This would allow a free app to gracefully upgrade if they didn't expect to become popular over night and were using some API that could get blocked. So hit ceiling price -> enter grace period -> pay or get blocked.

### PR DESCRIPTION
**Pull request by sanderpick**
Closes #390

- Better stored data tracking inside the buckets service
- Adds a `FreeQuotaGracePeriod` config param to `billingd`

### CODE REVIEW

File: api/billingd/client/client_test.go [hunk 1]
```diff
@@ -281,10 +281,11 @@ func setup(t *testing.T) *client.Client {
...
 		StripeAPIKey:           os.Getenv("STRIPE_API_KEY"),
 		StripeSessionReturnURL: "http://127.0.0.1:8006/dashboard",
-		DBURI:           "mongodb://127.0.0.1:27017",
-		DBName:          util.MakeToken(8),
-		GatewayHostAddr: util.MustParseAddr(fmt.Sprintf("/ip4/127.0.0.1/tcp/%d", gwPort)),
-		Debug:           true,
+		DBURI:                "mongodb://127.0.0.1:27017",
+		DBName:               util.MakeToken(8),
+		GatewayHostAddr:      util.MustParseAddr(fmt.Sprintf("/ip4/127.0.0.1/tcp/%d", gwPort)),
+		FreeQuotaGracePeriod: 0,
```
Comment (sanderpick) [inline api/billingd/client/client_test.go:287]: `0` grace period for tests

File: api/billingd/main.go [hunk 2]
```diff
@@ -63,6 +64,10 @@ var (
 				Key:      "stripe.webhook_secret",
 				DefValue: "",
 			},
+			"freeQuotaGracePeriod": {
+				Key:      "free_quota_grace_period",
+				DefValue: time.Hour * 24 * 7,
```
Comment (sanderpick) [inline api/billingd/main.go:69]: 1 week by default

File: api/billingd/service/service.go [hunk 3]
```diff
@@ -806,10 +809,18 @@ func (s *Service) handleUsage(ctx context.Context, cus *Customer, product Produc
...
-		return nil, common.ErrExceedsFreeQuota
+		now := time.Now().Unix()
+		if cus.GracePeriodStartAt == 0 {
+			cus.GracePeriodStartAt = now
+			update["grace_period_start_at"] = cus.GracePeriodStartAt
+		}
+		deadline := cus.GracePeriodStartAt + int64(s.config.FreeQuotaGracePeriod.Seconds())
+		if now >= deadline {
+			return nil, common.ErrExceedsFreeQuota
+		}
```
Comment (sanderpick) [inline api/billingd/service/service.go:822]: Grace period starts when the user initially goes over the limit. This does not account for the case where they might drop down again. Could add that later, but this is good nuff for now. Also allows for a `0` grace period.

File: core/usage_interceptor.go [hunk 4]
```diff
@@ -136,7 +137,7 @@ func (t *Textile) preUsageFunc(ctx context.Context, method string) (context.Cont
 			StorageUsed: cus.DailyUsage["stored_data"].Total,
 		}
 		if cus.Billable {
-			owner.StorageAvailable = -1
+			owner.StorageAvailable = int64(math.MaxInt64)
```
Comment (jsign) [inline core/usage_interceptor.go:140]: Nice, simplifies ifs cases.

**Review by jsign — APPROVED** [review]
LGTM.

**Pull request merged**